src/backend/common/queries/dict_converters/converter_base.py

```diff
@@ -0,0 +1,63 @@
+import abc
+from typing import Generic, List, Union
+
+from backend.common.queries.types import QueryReturn
+from backend.common.profiler import Span
+
+
+class ConverterBase(Generic[QueryReturn]):
+    _query_return: QueryReturn
+
```

Comment:
Maybe we can promote the versioning to first class?

```
SubversionMap = TypedDict("SubversionMap", {f"{v}": int for v in list(ApiMajorVersion)})

@abc.abstractproperty
SUBVERSIONS: SubversionMap
```

And this way we can make the typechecker tell us everywhere we'd need to add a new set of versions if/when we add apiv4